src/yetibot/commands/aws/formatters.clj

```diff
@@ -105,79 +109,122 @@
                            request-id)}))
 
-(defmethod format-response ::IAMAccessKeyCreated
+(defmethod format-iam-response ::IAMAccessKeyCreated
   [{{:keys [UserName AccessKeyId Status SecretAccessKey CreateDate]} :AccessKey}]
   {:result/data  {:user-name UserName, :access-key-id AccessKeyId, :status Status, :secret-access-key SecretAccessKey, :create-date CreateDate}
    :result/value (format "An access key for user %s has been successfully created on %s\nAWS Access Key ID : %s\nAWS Secret Access Key : %s\nStatus : %s\n"
                          UserName CreateDate AccessKeyId SecretAccessKey Status)})
 
-(defmethod format-response ::IAMListAccessKeysResponseReceived
+(defmethod format-iam-response ::IAMListAccessKeysResponseReceived
   [{:keys [AccessKeyMetadata]}]
   {:result/data  {:access-key-metadata AccessKeyMetadata}
    :result/value (map #(format "Access key ID : %s\nStatus : %s\nCreated on : %s\n"
                                (:AccessKeyId %) (:Status %) (:CreateDate %))
                       AccessKeyMetadata)})
 
-(defmethod format-response ::IAMAccessKeyDeleted
+(defmethod format-iam-response ::IAMAccessKeyDeleted
   [response]
   (let [request-id (get-in response [:DeleteAccessKeyResponse :ResponseMetadata :RequestId])]
     {:result/data  {:request-id request-id}
      :result/value (format "Access Key successfully deleted [RequestId=%s]"
-                           request-id)}))
\ No newline at end of file
+                           request-id)}))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defmulti format-s3-response
+          "Returns a dispatch-value matching the S3 operation that has been successfully invoked
+          or has failed"
+          (fn [response]
+            (let [aws-type (:aws/type (meta response))]
+              (cond
+                (and (s/valid? (s3-response-spec :CreateBucket) response)
+                     (= aws-type :aws.type/CreateBucket)
+                     (not (contains? response :Error))) ::S3BucketCreated
+                (and (s/valid? (s3-response-spec :ListBuckets) response)
+                     (= aws-type :aws.type/ListBuckets)
+                     (not (contains? response :Error))) ::S3BucketListed
+                (and (s/valid? (s3-response-spec :ListObjectsV2) response)
+                     (= aws-type :aws.type/ListObjects)
+                     (not (contains? response :Error))) ::S3ObjectsListed
+                :else ::error))))
+
+(defmethod format-s3-response ::error
+  [response]
+  {:result/error (get-in response [:Error :Message])})
+
+(defmethod format-s3-response ::S3BucketCreated
+  [{:keys [Location]}]
+  {:result/data  {:location Location}
+   :result/value (format "S3 bucket successfully created at %s" Location)})
+
+(defmethod format-s3-response ::S3BucketListed
+  [{:keys [Buckets]}]
+  {:result/data  {:buckets Buckets}
```

Comment:
Same - more data?

Reply:
For the _bucket list_, we have more _infos_ related to the buckets. I will paste the _response format_ here and we will decide which data we will include in the _formatted_ response.

Reply:
We have the following _spec_ for the `:ListBuckets` API call : 

```
:ListBuckets {:name "ListBuckets",
               :documentation "<p>Returns a list of all buckets owned by the authenticated sender of the request.</p>",
               :documentationUrl "http://docs.amazonwebservices.com/AmazonS3/latest/API/RESTServiceGET.html",
               :request nil,
               :required nil,
               :response {:Buckets [:seq-of {:Name string, :CreationDate timestamp}],
                          :Owner {:DisplayName string, :ID string}}},
```
Apart from the _Buckets list_, we have an additional `:Owner` field included.
Do you want it to be included within the response @devth ?

Reply:
Sure. Can't hurt to have extra data behind the pipe :)